src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseSimulationResource.java

```diff
@@ -0,0 +1,190 @@
+        }
+
+        // Check if programming language is set
+        if (programmingExercise.getProgrammingLanguage() == null) {
+            return ResponseEntity.badRequest().headers(HeaderUtil.createAlert(applicationName, "No programming language was specified", "programmingLanguageNotSet")).body(null);
+        }
+
+        // Check if package name is set
+        if (programmingExercise.getProgrammingLanguage() == ProgrammingLanguage.JAVA) {
+            // only Java needs a valid package name at the moment
+            if (programmingExercise.getPackageName() == null || programmingExercise.getPackageName().length() < 3) {
+                return ResponseEntity.badRequest().headers(HeaderUtil.createAlert(applicationName, "The package name is invalid", "packagenameInvalid")).body(null);
+            }
+
+            // Check if package name matches regex
+            Matcher packageNameMatcher = packageNamePattern.matcher(programmingExercise.getPackageName());
+            if (!packageNameMatcher.matches()) {
+                return ResponseEntity.badRequest().headers(HeaderUtil.createAlert(applicationName, "The package name is invalid", "packagenameInvalid")).body(null);
+            }
+        }
+
+        // Check if max score is set
+        if (programmingExercise.getMaxScore() == null) {
+            return ResponseEntity.badRequest().headers(HeaderUtil.createAlert(applicationName, "The max score is invalid", "maxscoreInvalid")).body(null);
+        }
+
+        programmingExercise.generateAndSetProjectKey();
+        try {
+            ProgrammingExercise newProgrammingExercise = programmingExerciseSimulationService.setupProgrammingExerciseWithoutLocalSetup(programmingExercise); // Setup all
+                                                                                                                                                              // repositories etc
+            programmingExerciseSimulationService.setupInitialSubmissionsAndResults(programmingExercise);
+            return ResponseEntity.created(new URI("/api/programming-exercises" + newProgrammingExercise.getId()))
+                    .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, ENTITY_NAME, newProgrammingExercise.getTitle())).body(newProgrammingExercise);
+        }
+        catch (URISyntaxException e) {
+            log.error("Error while setting up programming exercise", e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+                    .headers(HeaderUtil.createAlert(applicationName, "An error occurred while setting up the exercise: " + e.getMessage(), "errorProgrammingExercise")).body(null);
+        }
+    }
+
+    public static final class Endpoints {
+
+        public static final String ROOT = "/api";
+
+        public static final String PROGRAMMING_EXERCISES = "/programming-exercises";
+
+        public static final String NO_LOCAL_SETUP = PROGRAMMING_EXERCISES + "/no-local-setup";
```

Comment:
I don't really like this name. Either call it "local-setup" or something with "simulation"

Reply:
i renamed the endpoint, now it is called EXERCISES_SIMULATION